tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -298,12 +328,127 @@ private void CancelButtonPressed()
         IsCancelling = true;
     }
 
+    /// <summary>
+    /// The accounts the user is logged into is stored here.
+    /// </summary>
+    [ObservableProperty]
+    private MenuFlyout _accountsToShow;
+
+    /// <summary>
+    /// Switches the repos shown to the account selected.
+    /// </summary>
+    /// <param name="sender">The object sending the event.</param>
+    /// <param name="e">The arguments</param>
+    private async void MenuItemClick(object sender, RoutedEventArgs e)
+    {
+        if (sender is MenuFlyoutItem selectedItem)
+        {
+            SelectedAccount = selectedItem.Text;
+            await GetRepositoriesAsync(_selectedRepoProvider, SelectedAccount);
+
+            var sdkDisplayName = _providers.GetSDKProvider(_selectedRepoProvider).DisplayName;
+            _addRepoDialog.SelectRepositories(SetRepositories(sdkDisplayName, SelectedAccount));
+        }
+    }
+
+    /// <summary>
+    /// Makes the MenuFlyout object used to display multple accounts in the repo tool.
+    /// </summary>
+    /// <returns>The MenuFlyout to display.</returns>
+    /// <remarks>
+    /// The MenuFlyout is constructed in code to apply the correct callback methods on the Menu Items.
+    /// </remarks>
+    private MenuFlyout ConstructFlyout()

```

Comment:
Code-behind is be a better candidate for this method

Reply:
It'll stay where it is right now because moving the code from the view-model to code behind requires moving other code.  But, why do you say so?  Because the code effects the UI?

Reply:
Yes, this is UI logic